### DISCUSSION TITLE
Is there a way to use message retraction?

### DISCUSSION THREAD

**Discussion by armor009** [processone/ejabberd#3640] [Questions] [answered]
Is there a way to use message retraction?
https://xmpp.org/extensions/xep-0424.html

**Comment by badlop** [comment]
There is no way to use message retraction. It isn't implemented in ejabberd, and I am not aware of any additional module that implements it.

**Comment by Neustradamus** [comment]
@badlop: For example, it is in ConverseJS, MongooseIM, Movim, Slixmpp and Smack:
- https://github.com/poezio/slixmpp/blob/master/docs/api/plugins/xep_0424.rst
- https://github.com/igniterealtime/Smack/blob/master/documentation/extensions/index.md
- https://github.com/conversejs/converse.js/blob/master/README.md
- https://github.com/movim/movim/blob/master/CHANGELOG.md
- https://github.com/esl/MongooseIM/blob/master/CHANGELOG.md

**Comment by badlop** [comment, marked as answer]
Good news! Support for [XEP-0424 Message Retraction](https://xmpp.org/extensions/xep-0424.html) has been just implemented and [committed to ejabberd git](https://github.com/processone/ejabberd/commit/a4bb695fc3e694622f443fa040c0add89ee806f1), and will be included in the next ejabberd release.
